### PR TITLE
Added getValue method for all non-QtiScalar datatypes.

### PR DESCRIPTION
**Pull request by julien-sebire**
This PR allows the usage of getValue method for all non-QtiScalar data types, using the implemented __toString method of each of these types.

The implementation is made with a QtiNonScalar interface and a QtiNonScalarTrait for holding the code, since QtiCoords extends IntegerCollection, it cannot extend another class, hence the usage of a trait for a common addition in all non-QtiScalar types.

### CODE REVIEW

File: src/qtism/runtime/pci/json/Marshaller.php [hunk 1]
```diff
@@ -136,7 +136,7 @@ protected function marshallUnit($unit)
 
             foreach ($unit as $u) {
                 $data = $this->marshallUnit($u);
-                $json['list'][$strBaseType][] = $data['base'][$strBaseType];
+                $json['list'][$strBaseType][] = $data['base'][$strBaseType] ?? null;
```
Comment (bugalot) [inline src/qtism/runtime/pci/json/Marshaller.php:139]: What does that solve? Seems out of scope of the PR

File: src/qtism/data/content/interactions/GraphicOrderInteraction.php [hunk 2]
```diff
@@ -232,7 +232,7 @@ public function getResponseValidityConstraint()
         return new ResponseValidityConstraint(
             $this->getResponseIdentifier(),
             ($this->hasMinChoices() === true) ? $this->getMinChoices() : count($this->getHotspotChoices()),
-            ($this->hasMinChoices() === false) ? 0 : ($this->hasMaxChoices() === true) ? $this->getMaxChoices() : 0
+            $this->hasMinChoices() && $this->hasMaxChoices() ? $this->getMaxChoices() : 0
```
Comment (bugalot) [inline src/qtism/data/content/interactions/GraphicOrderInteraction.php:235]: Please double check it I feel it does not support all cases anymore.

**Comment by julien-sebire** [comment]
Replaced by https://github.com/oat-sa/qti-sdk/pull/187
